src/pages/collaborations/collaborationManagementForm.jsx

```diff
@@ -134,23 +170,37 @@ export default function CollaborationManagementForm({
           loading={loading}
           onClick={async () => {
             if (
-              !collaborationAlreadyExists(
+              collaborationsContainsAlreadyMutuallyRevokedCollab(
                 existingCollaborations,
                 user1,
                 user2,
               )
             ) {
-              await establishCollaboration(
-                // need the await here. Otherwise, setShouldDisplay(true) below fires before this completes
-                user1,
-                user2,
-              );
-            } else {
               setError(
                 intl.formatMessage({
-                  id: 'COLLABORATION_ALREADY_EXISTS',
+                  id: 'REVOKED_COLLAB_EXISTS',
                 }),
               );
+            } else {
+              if (
+                !collaborationAlreadyExists(
+                  existingCollaborations,
+                  user1,
+                  user2,
+                )
+              ) {
+                await establishCollaboration(
+                  // need the await here. Otherwise, setShouldDisplay(true) below fires before this completes
```

Comment:
you need the await but you dont need the comment :)

Reply:
https://media.giphy.com/media/1qWIZcX7lrhRKT7GIK/giphy.gif